include/aikido/planner/WorldStateSaver.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef AIKIDO_PLANNER_WORLDSTATESAVER_HPP_
+#define AIKIDO_PLANNER_WORLDSTATESAVER_HPP_
+#include "World.hpp"
+
+namespace aikido {
+namespace planner {
+
+/// RAII class to save and restore a World's state.
+class WorldStateSaver
+{
+public:
+  /// Construct a WorldStateSaver and save the current state of the
+  /// \c World. This state will be restored when
+  /// WorldStateSaver is destructed.
+  ///
+  /// \param _space WorldState to save/restore
+  explicit WorldStateSaver(WorldPtr world);
```

Comment:
We should be able to flag this argument as `const`.

Also, I don't see a strong reason to make this accept a smart pointer rather than a reference or raw pointer.